### ISSUE TITLE
Unary tests behave different when evaluated directly or via an expression using the keyword "in"

### ISSUE THREAD

**Issue by till-stadtler** [camunda/feel-scala#768]
**Describe the bug**
The following unary test, which in itself does not make a lot of sense, behaves differently when being evaluated as a unary test and when being evaluated via an expression using the keyword "in":
```
Input value: 5
Unary test: not(7)
// returns: true
// expected: ?, but the same result as the expression below

Expressions: 5 in not(7)
// returns: false
// expected: ?, but same result as the unary test above
```

Further examination:
`not(7)` returns null because the negation expects a boolean.
The unary test:
```
Input value: 5
Unary test: null
```
returns `false`. 
This is the same behavior as the expression `5 in null`.

So, the faulty behavior might be the unary test
```
Input value: 5
Unary test: not(7)
```
returning `true`. (?)

**To Reproduce**
Enter the expressions above in the FEEL Playground.

**Expected behavior**
Unary tests and expressions using the keyword "in" should behave the same.

**Environment**
* FEEL engine version: latest (deployed to Playground)
* Affects: Unclear

**Comment by saig0** [comment]
`not()` is a special case because it is ambiguous. 
- [not()](https://docs.camunda.io/docs/components/modeler/feel/language-guide/feel-unary-tests/#negationnot) for unary-tests 
- [not()](https://docs.camunda.io/docs/components/modeler/feel/builtin-functions/feel-built-in-functions-boolean/#notnegand) as a built-in function

@till-stadtler do you see other examples that don't use the `not()` function?

**Comment by till-stadtler** [comment]
@saig0, no, I have not found any other examples for expressions with "in" and unary tests behaving differently.

So, regarding this bug ticket, when not() is used after "in" in an expression, it should be the not() for unary-tests, but currently is evaluated as not() as a built-in function?

**Comment by saig0** [comment]
> I have not found any other examples for expressions with "in" and unary tests behaving differently.

Thank you for checking. :+1: So, we can make the issue more concrete and focus on the combination of `.. in not(..)`

> when not() is used after "in" in an expression, it should be the not() for unary-tests, but currently is evaluated as not() as a built-in function?

Currently, it uses the `not()` boolean built-in function. 

The expected behavior is unclear. From the FEEL grammar, `in` is followed by a `positive unary test` that doesn't include `not()` from the `unary tests`.

To do: 
- Scan the DMN specification for the usage of `in`
- Check the DMN TCK for test cases
